### DISCUSSION TITLE
Gradient checkpointing/ rematerialization

### DISCUSSION THREAD

**Discussion by jakubMitura14** [LuxDL/Lux.jl#817] [Q&A] [answered]
Hello is there a way for gradient checkpointing so instead of saving values of forward pass during backward pass recalculate forward function to reduce memory consumption of lux layer

**Comment by avik-pal** [comment, marked as answer]
For Zygote you can use https://fluxml.ai/Zygote.jl/stable/utils/#Zygote.checkpointed.

I am not sure if Enzyme has one built in (@wsmoses might know?), though mimicking the Zygote one for Enzyme shouldn't be hard.
